File: java/src/main/java/com/openalpr/api/models/AlprPlate.java
package com.openalpr.api.models;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import com.openalpr.api.models.PlateDetails;
import com.openalpr.api.models.RegionOfInterest;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.math.BigDecimal;
import java.util.*;



/**
 * Results from plate analysis task
 **/

@ApiModel(description = "Results from plate analysis task")
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-02-17T14:18:32.383-05:00")
public class AlprPlate   {
  
  private List<PlateDetails> results = new ArrayList<PlateDetails>();
  private Integer imgWidth = null;
  private Integer imgHeight = null;
  private List<RegionOfInterest> regionsOfInterest = new ArrayList<RegionOfInterest>();
  private BigDecimal epochTime = null;
  private Integer version = null;


  public enum DataTypeEnum {
    ALPR_RESULTS("alpr_results"),
    ALPR_GROUP("alpr_group"),
    HEARTBEAT("heartbeat");

    private String value;

    DataTypeEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return value;
    }
  }

  private DataTypeEnum dataType = null;
  private BigDecimal processingTimeMs = null;

  
  /**
   **/
  
  @ApiModelProperty(value = "")
  @JsonProperty("results")
  public List<PlateDetails> getResults() {
    return results;
  }
  public void setResults(List<PlateDetails> results) {
    this.results = results;
  }

  
  /**
   * Width of the uploaded image in pixels
   **/
  
  @ApiModelProperty(value = "Width of the uploaded image in pixels")
  @JsonProperty("img_width")
  public Integer getImgWidth() {
    return imgWidth;
  }
  public void setImgWidth(Integer imgWidth) {
    this.imgWidth = imgWidth;
  }

  
  /**
   * Height of the input image in pixels
   **/
  
  @ApiModelProperty(value = "Height of the input image in pixels")
  @JsonProperty("img_height")
  public Integer getImgHeight() {
    return imgHeight;
  }
  public void setImgHeight(Integer imgHeight) {
    this.imgHeight = imgHeight;
  }

  
  /**
   * Describes the areas analyzed in the input image
   **/
  
  @ApiModelProperty(value = "Describes the areas analyzed in the input image")
  @JsonProperty("regions_of_interest")
  public List<RegionOfInterest> getRegionsOfInterest() {
    return regionsOfInterest;
  }
  public void setRegionsOfInterest(List<RegionOfInterest> regionsOfInterest) {
    this.regionsOfInterest = regionsOfInterest;
  }

  
  /**
   * Epoch time that the image was processed in milliseconds
   **/
  
  @ApiModelProperty(value = "Epoch time that the image was processed in milliseconds")
  @JsonProperty("epoch_time")
  public BigDecimal getEpochTime() {
    return epochTime;
  }
  public void setEpochTime(BigDecimal epochTime) {
    this.epochTime = epochTime;
  }

  
  /**
   * API format version
   **/
  
  @ApiModelProperty(value = "API format version")
  @JsonProperty("version")
  public Integer getVersion() {
    return version;
  }
  public void setVersion(Integer version) {
    this.version = version;
  }

  
  /**
   * Specifies the type of data in this response
   **/
  
  @ApiModelProperty(value = "Specifies the type of data in this response")
  @JsonProperty("data_type")
  public DataTypeEnum getDataType() {
    return dataType;
  }
  public void setDataType(DataTypeEnum dataType) {
    this.dataType = dataType;
  }

  
  /**
   * Number of milliseconds to process all license plates
   **/
  
  @ApiModelProperty(value = "Number of milliseconds to process all license plates")
  @JsonProperty("processing_time_ms")
  public BigDecimal getProcessingTimeMs() {
    return processingTimeMs;
  }
  public void setProcessingTimeMs(BigDecimal processingTimeMs) {
    this.processingTimeMs = processingTimeMs;
  }

  

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AlprPlate alprPlate = (AlprPlate) o;
    return Objects.equals(results, alprPlate.results) &&
        Objects.equals(imgWidth, alprPlate.imgWidth) &&
        Objects.equals(imgHeight, alprPlate.imgHeight) &&
        Objects.equals(regionsOfInterest, alprPlate.regionsOfInterest) &&
        Objects.equals(epochTime, alprPlate.epochTime) &&
        Objects.equals(version, alprPlate.version) &&
        Objects.equals(dataType, alprPlate.dataType) &&
        Objects.equals(processingTimeMs, alprPlate.processingTimeMs);
  }

  @Override
  public int hashCode() {
    return Objects.hash(results, imgWidth, imgHeight, regionsOfInterest, epochTime, version, dataType, processingTimeMs);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AlprPlate {\n");
    
    sb.append("    results: ").append(toIndentedString(results)).append("\n");
    sb.append("    imgWidth: ").append(toIndentedString(imgWidth)).append("\n");
    sb.append("    imgHeight: ").append(toIndentedString(imgHeight)).append("\n");
    sb.append("    regionsOfInterest: ").append(toIndentedString(regionsOfInterest)).append("\n");
    sb.append("    epochTime: ").append(toIndentedString(epochTime)).append("\n");
    sb.append("    version: ").append(toIndentedString(version)).append("\n");
    sb.append("    dataType: ").append(toIndentedString(dataType)).append("\n");
    sb.append("    processingTimeMs: ").append(toIndentedString(processingTimeMs)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: javascript/src/model/AlprPlate.js
// require files in Node.js environment
var PlateDetails;var RegionOfInterest;
if (typeof module === 'object' && module.exports) {
  
  PlateDetails = require('./PlateDetails.js');
  RegionOfInterest = require('./RegionOfInterest.js');
}



//export module
if ( typeof define === "function" && define.amd ) {     
	define('DataTypeEnum', ['jquery'], function($) {
        return DataTypeEnum;
	 });
}

var DataTypeEnum = function DataTypeEnum() {
	var self = this;
	

	/**
	 * @const 
	 */ 
	self.ALPR_RESULTS = "alpr_results",
	
	/**
	 * @const 
	 */ 
	self.ALPR_GROUP = "alpr_group",
	
	/**
	 * @const 
	 */ 
	self.HEARTBEAT = "heartbeat";

}


//export module
if ( typeof define === "function" && define.amd ) {
  define('AlprPlate', ['jquery', 'Array', 'Array'],
    function($, Array, Array) {
      return AlprPlate;
   });
}

/**
 * Results from plate analysis task
 **/
var AlprPlate = function AlprPlate() { 
  var self = this;
  
  /**
   * datatype: Array
   **/
  self.results = [];
  
  /**
   * Width of the uploaded image in pixels
   * datatype: Integer
   **/
  self.imgWidth = null;
  
  /**
   * Height of the input image in pixels
   * datatype: Integer
   **/
  self.imgHeight = null;
  
  /**
   * Describes the areas analyzed in the input image
   * datatype: Array
   **/
  self.regionsOfInterest = [];
  
  /**
   * Epoch time that the image was processed in milliseconds
   * datatype: Number
   **/
  self.epochTime = null;
  
  /**
   * API format version
   * datatype: Integer
   **/
  self.version = null;
  
  /**
   * Specifies the type of data in this response
   * datatype: DataTypeEnum
   **/
  self.dataType = null;
  
  /**
   * Number of milliseconds to process all license plates
   * datatype: Number
   **/
  self.processingTimeMs = null;
  
  
  self.constructFromObject = function(data) {
    if (!data) {
      return;
    }
    
    self.results = new Array();
    
    self.imgWidth = data.img_width;
    
    self.imgHeight = data.img_height;
    
    self.regionsOfInterest = new Array();
    
    self.epochTime = data.epoch_time;
    
    self.version = data.version;
    
    self.dataType = data.data_type;
    
    self.processingTimeMs = data.processing_time_ms;
    
  }

  
  /**
   * @return {Array}
   **/
  self.getResults = function() {
    return self.results;
  }

  /**
   * @param {Array} results
   **/
  self.setResults = function (results) {
    self.results = results;
  }
  
  /**
   * get Width of the uploaded image in pixels
   * @return {Integer}
   **/
  self.getImgWidth = function() {
    return self.imgWidth;
  }

  /**
   * set Width of the uploaded image in pixels
   * @param {Integer} imgWidth
   **/
  self.setImgWidth = function (imgWidth) {
    self.imgWidth = imgWidth;
  }
  
  /**
   * get Height of the input image in pixels
   * @return {Integer}
   **/
  self.getImgHeight = function() {
    return self.imgHeight;
  }

  /**
   * set Height of the input image in pixels
   * @param {Integer} imgHeight
   **/
  self.setImgHeight = function (imgHeight) {
    self.imgHeight = imgHeight;
  }
  
  /**
   * get Describes the areas analyzed in the input image
   * @return {Array}
   **/
  self.getRegionsOfInterest = function() {
    return self.regionsOfInterest;
  }

  /**
   * set Describes the areas analyzed in the input image
   * @param {Array} regionsOfInterest
   **/
  self.setRegionsOfInterest = function (regionsOfInterest) {
    self.regionsOfInterest = regionsOfInterest;
  }
  
  /**
   * get Epoch time that the image was processed in milliseconds
   * @return {Number}
   **/
  self.getEpochTime = function() {
    return self.epochTime;
  }

  /**
   * set Epoch time that the image was processed in milliseconds
   * @param {Number} epochTime
   **/
  self.setEpochTime = function (epochTime) {
    self.epochTime = epochTime;
  }
  
  /**
   * get API format version
   * @return {Integer}
   **/
  self.getVersion = function() {
    return self.version;
  }

  /**
   * set API format version
   * @param {Integer} version
   **/
  self.setVersion = function (version) {
    self.version = version;
  }
  
  /**
   * get Specifies the type of data in this response
   * @return {DataTypeEnum}
   **/
  self.getDataType = function() {
    return self.dataType;
  }

  /**
   * set Specifies the type of data in this response
   * @param {DataTypeEnum} dataType
   **/
  self.setDataType = function (dataType) {
    self.dataType = dataType;
  }
  
  /**
   * get Number of milliseconds to process all license plates
   * @return {Number}
   **/
  self.getProcessingTimeMs = function() {
    return self.processingTimeMs;
  }

  /**
   * set Number of milliseconds to process all license plates
   * @param {Number} processingTimeMs
   **/
  self.setProcessingTimeMs = function (processingTimeMs) {
    self.processingTimeMs = processingTimeMs;
  }
  

  self.toJson = function () {
    return JSON.stringify(self);
  }
}

if (typeof module === 'object' && module.exports) {
  module.exports = AlprPlate;
}
